csharp/src/Microsoft.ML.OnnxRuntime/Training/TrainingSession.shared.cs

```diff
@@ -529,15 +530,15 @@ public FixedBufferOnnxValue ToBuffer(bool onlyTrainable)
         /// Loads the training session model parameters from a contiguous buffer
         /// </summary>
         /// <param name="buffer">Contiguous buffer to load the parameters from.</param>
-        public void FromBuffer(FixedBufferOnnxValue buffer)
+        public void FromBuffer(OrtValue buffer)
         {
-            if (buffer.OnnxValueType != OnnxValueType.ONNX_TYPE_TENSOR)
+            if (buffer.OnnxType != OnnxValueType.ONNX_TYPE_TENSOR)
             {
                 throw new ArgumentException("Incorrect buffer received. Expected a tensor buffer.");
             }
 
             IntPtr typeAndShapeInfo = IntPtr.Zero;
-            NativeApiStatus.VerifySuccess(NativeMethods.OrtGetTensorTypeAndShape(buffer.Value.Handle, out typeAndShapeInfo));
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtGetTensorTypeAndShape(buffer.Handle, out typeAndShapeInfo));
```

Comment:
>NativeApiStatus.VerifySuccess(NativeMethods.OrtGetTensorTypeAndShape(buffer.Handle, out typeAndShapeInfo)); [](http://example.com/codeflow?start=12&length=107)

Suggest using OrtValue public interface. This complicates refactoring should it become necessary

Reply:
Are you going to remove the not needed code?